src/components/encoding-pane/property-editor.tsx

```diff
@@ -14,26 +15,32 @@ export interface PropertyEditorProps extends ActionHandler<SpecEncodingAction> {
   fieldDef: ShelfFieldDef;
 }
 
-export class PropertyEditorBase extends React.PureComponent<PropertyEditorProps, {}> {
+export interface PropertyEditorState {
```

Comment:
How do you make sure that the component's internal state is in sync with the spec in the store?
(Imagine a case where you load a spec into Voyager and it already has log scale.) 

Instead of keeping internal state, better read directly from the store and derive a sensible `formData` from the stored value?  

That said, you know more about this infrastructure, so I might have overlooked something here.

Reply:
This comment makes sense. I'll try to address this in the next pass!